src/utils/displayname-integration.test.ts

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2025 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE in the repository root for full details.
+*/
+
+import { afterEach, beforeAll, describe, expect, test, vi } from "vitest";
+
+import { shouldDisambiguate } from "./displayname";
+import { alice } from "./test-fixtures";
+import { mockMatrixRoom } from "./test";
+
+// Ideally these tests would be in ./displayname.test.ts but I can't figure out how to
+// just spy on the removeHiddenChars() function without impacting the other tests.
+// So, these tests are in this separate test file.
```

Comment:
Is the issue here, that `removeHiddenChars` is already mocked in the other tests and you would need to restore them back to the previous mocked/spyOn implementation?

Otherwise, `jsUtils.removeHiddenChars.restoreMocks()` should do the trick

Reply:
The issue is that the other tests need the real implementation of the js-sdk for the tests to complete.

If I don't do `vi.mock()` then the spy on `removeHiddenChars` does not work.

If I do `vi.mock()` without implementation then the other tests fail.

If I do `vi.mock()` and provide an implementation of the mock using `vi.importActual()` then it fails in some way (I can't remember right now).

Reply:
I see vi.mock is the issue...